manager/allocator/network.go

```diff
@@ -371,12 +371,15 @@ func (a *Allocator) doNetworkAlloc(ctx context.Context, ev events.Event) {
 		s := v.Service.Copy()
 
 		if nc.nwkAllocator.IsServiceAllocated(s) {
-			break
-		}
-
-		if err := a.allocateService(ctx, s); err != nil {
-			log.G(ctx).WithError(err).Errorf("Failed allocation during update of service %s", s.ID)
-			break
+			if nc.nwkAllocator.PortsAllocatedInHostPublishMode(s) {
```

Comment:
@dongluochen The Host PublishMode was separated from `IsServiceAllocated` in #1691.

Not very familiar with this part of the code though I tend to think think one reason to avoid handling Host PublishMode was that,  services will be updated if `IsServiceAllocated` is false, which may be unnecessary for Host PublishMode.

/cc @aboch @mrjana